dev/com.ibm.ws.logging.osgi/src/com/ibm/ws/logging/internal/osgi/TrOSGiLogForwarder.java

```diff
@@ -98,10 +112,35 @@ OSGiTraceComponent getTraceComponent(Bundle b) {
         }
     }
 
-    @Override
+    private Collection<String> getEquinoxTraceGroups(Bundle b) {
+        // the .options resource contains all the trace group keys
+        // that Equinox uses.
+        Properties options = new Properties();
+        try {
+            options.load(b.getResource(".options").openStream());
+        } catch (IOException e) {
+            // auto FFDC
+        }
+        return options.stringPropertyNames();
+    }
+
+	@Override
     public void logged(LogEntry le) {
+        if (currentlyLogging.get() == Boolean.TRUE) {
```

Comment:
When this method recurses do we know anything about the ignored LogEvent that could be documented here?

Reply:
Not really, but not sure what we can do about it without risking more recursion.

Reply:
Got it.  My thinking for that obscure question was, best case, LogEvent might be the same object that was passed to the previous call to logged() on the same thread.  Then, the LogEvent wouldn't matter.  Or, the LogEvent is always the same when the recursion occurs and we could document the condition.